src/parser/deathknight/unholy/modules/features/FesteringStrike.tsx

```diff
@@ -17,49 +18,56 @@ class FesteringStrike extends Analyzer {
     this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.FESTERING_STRIKE), this.onCast);
   }
 
   // used to track how many stacks a target has
-  targets = {};
-
+  targets: number[] = [];
   totalFesteringStrikeCasts = 0;
   festeringStrikeCastsOverThreeStacks = 0;
 
-  onWoundApply(event){
+  onWoundApply(event: ApplyDebuffStackEvent){
     // we only need to look at events applying stacks after the first one, our analysis only cares about times when stacks are >3
     this.targets[event.targetID] = event.stack;
   }
 
-  onWoundRemove(event){
-    this.targets[event.targetID] = event.stack || 0;
+  onWoundRemove(event: RemoveDebuffStackEvent){
+    this.targets[event.targetID] = event.stack;
+  }
+
+  onAllWoundRemove(event: RemoveDebuffEvent){
+    this.targets[event.targetID] = 0;
   }
 
-  onCast(event){
-      this.totalFesteringStrikeCasts += 1;
-      if(this.targets[event.targetID]){
-        const currentTargetWounds = this.targets[event.targetID];
-        if(currentTargetWounds > 3){
-          this.festeringStrikeCastsOverThreeStacks += 1;
-        }
+  onCast(event: CastEvent){
+
+    this.totalFesteringStrikeCasts += 1;
+    if(this.targets[event.targetID!]){
+      const currentTargetWounds = this.targets[event.targetID!];
+      if(currentTargetWounds > 3){
+        this.festeringStrikeCastsOverThreeStacks += 1;
       }
+    }
   }
 
-  suggestions(when) {
+  suggestions(when: When) {
     const percentCastsOverThreeStacks = this.festeringStrikeCastsOverThreeStacks/this.totalFesteringStrikeCasts;
     const strikeEfficiency = 1 - percentCastsOverThreeStacks;
     when(strikeEfficiency).isLessThan(0.60)
-        .addSuggestion((suggest, actual, recommended) => suggest(<span>You are casting <SpellLink id={SPELLS.FESTERING_STRIKE.id} /> too often.  When spending runes remember to cast <SpellLink id={SPELLS.SCOURGE_STRIKE.id} /> instead on targets with more than three stacks of <SpellLink id={SPELLS.FESTERING_WOUND.id} /></span>)
-            .icon(SPELLS.FESTERING_STRIKE.icon)
-            .actual(`${formatPercentage(actual)}% of Festering Strikes did not risk overcapping Festering Wounds`)
-            .recommended(`>${formatPercentage(recommended)}% is recommended`)
-            .regular(recommended - 0.30).major(recommended - 0.40));
+      .addSuggestion((suggest, actual, recommended) => suggest(<span>You are casting <SpellLink id={SPELLS.FESTERING_STRIKE.id} /> too often.  When spending runes remember to cast <SpellLink id={SPELLS.SCOURGE_STRIKE.id} /> instead on targets with more than three stacks of <SpellLink id={SPELLS.FESTERING_WOUND.id} /></span>)
+      .icon(SPELLS.FESTERING_STRIKE.icon)
+      .actual(`${formatPercentage(actual)}% of Festering Strikes did not risk overcapping Festering Wounds`)
+      .recommended(`>${formatPercentage(recommended)}% is recommended`)
+      .regular(recommended - 0.30).major(recommended - 0.40));
   }
```

Comment:
You should make a suggestion threshold for this like we have elsewhere. 
Something like this: 
```js
get strikeEfficiency() {
  return {
    actual: 1 - ((this.festeringStrikeCastsOverThreeStacks/this.totalFesteringStrikeCasts) || 0)
    isLessThan: {
      minor: xx,
      average: xx,
      major: xx,
    },
    style: ThresholdStyle.X,
  };
}
```

Then you can remove the .regular and .major chained at the end, as well as remove the magic numbers